clang/tools/clang-scan-deps/ClangScanDeps.cpp

```diff
@@ -601,13 +601,20 @@ static bool useCAS() {
 }
 
 template <typename Container>
-static auto toJSONStrings(llvm::json::OStream &JOS, Container &&Strings) {
-  return [&JOS, Strings = std::forward<Container>(Strings)] {
+static auto toJSONStrings(llvm::json::OStream &JOS, Container &&Strings,
+                          bool ArePaths = false) {
+  return [&JOS, Strings = std::forward<Container>(Strings), ArePaths] {
     for (StringRef Str : Strings)
       // Not reporting SDKSettings.json so that test checks can remain (mostly)
       // platform-agnostic.
       if (!Str.ends_with("SDKSettings.json"))
-        JOS.value(Str);
+        if (ArePaths) {
+          llvm::SmallString<261> Path = Str;
+          llvm::sys::path::make_preferred(Path);
+          JOS.value(Path.str());
+        } else {
+          JOS.value(Str);
+        }
```

Comment:
```suggestion
if (Str.ends_with("SDKSettings.json"))
  continue;
if (Paths) {
  llvm::SmallString<261> Path{Str};
  llvm::sys::path::make_preferred(Path);
  JOS.value(Path.str());
} else {
  JOS.value(Str);
}
```

Reduces some of the indentation (and does happen to rename `ArePaths` to `Paths`.

Reply:
Done